### ISSUE TITLE
show more control plane info in OverView page

### ISSUE THREAD

**Issue by pantianying** [kiali/kiali#6177]
We are using the openshift service mesh and kiali, and now I have encountered a feature to add some monitoring information from the control surface to the overview page. The current page is as follows, and only a chart of istiod cpu information seems to be displayed.
<img width="1429" alt="image" src="https://github.com/kiali/kiali/assets/20589945/3d5267f1-42ad-48d1-80f3-4fe9ad7e671b">
I went to kiali's code and found that the other metric code was displayed in front of kiali. Use 'ShowMetrics' to determine if there is any data to show.
![image](https://github.com/kiali/kiali/assets/20589945/c6071316-5be6-4b35-9bfa-0a6d4ad61af4)

So is there a configuration that allows the overview page to display more information about the control panel? Such as memory information?

**Comment by nrfox** [comment]
Normally you should see memory information there alongside CPU. Two possible causes that I can think of for why you wouldn't see memory are 1. No memory metrics are being emitted or 2. The istiod deployment is missing the resource requirements for memory. Can you check both of those?

**Comment by pantianying** [comment]
> Normally you should see memory information there alongside CPU. Two possible causes that I can think of for why you wouldn't see memory are 1. No memory metrics are being emitted or 2. The istiod deployment is missing the resource requirements for memory. Can you check both of those?

yes ,  I checked and found that some metric such as container_memory_working_set_bytes were missing. but I installed using the standard openshift service mesh version.  and i have seen other screenshots with only one istiod cpu chart. So now the default is to show only one chart in the current version？

**Comment by leandroberetta** [comment]
@pantianying Hi! Let me take a look into this, because I've noticed in the past the same thing with OpenShift, I don't think it's a problem of your installation of Service Mesh. I think the issue is a missing metric in OpenShift. I'll let you know, I will check why it might be missing.

**Comment by pantianying** [comment]
> @pantianying Hi! Let me take a look into this, because I've noticed in the past the same thing with OpenShift, I don't think it's a problem of your installation of Service Mesh. I think the issue is a missing metric in OpenShift. I'll let you know, I will check why it might be missing.

Looking forward to your reply~

**Comment by leandroberetta** [comment]
Just to document this somewhere, this issue is not related to OpenShift Service Mesh as I installed Istio/Kiali on OpenShift and the memory chart doesn't show up:

![image](https://github.com/kiali/kiali/assets/1286393/79ecc268-9c8e-4e6a-ac12-70b9db34a1b5)

So, this might be a problem with the metric missing from OpenShift, probably because different container engines use different metrics?. Not sure, will keep investigating.

**Comment by pantianying** [comment]
> @pantianying after some research we noticed that the Prometheus that OpenShift Service Mesh is creating, does not scrape for the metrics related to the nodes (container_*) so there aren't present.
> 
> I thought about finding another metric that exists there, but the actual metric is the indicated for the job, so there isn't much to do there.
> 
> Is there any possibility for you to add that config to the Prometheus to start scraping for the metric?. I know that in the upcoming release of OSSM, you'll be able to use OpenShift's Prometheus (which has the metric, I checked that) or a custom one provided by you. The release should be out soon. That is another option, just wait a little bit to configure OSSM in that way.
> 
> One action I'll do is creating a FAQ entry to explaing this for the community.

Thank you for your reply. Do you mean that the problem can be solved by simply modifying the code? In fact, the kiali I use is based on tag
Modified code on v1.57. https://github.com/pantianying/kiali/tree/v1.57
So it's convenient for me to change the prometheus metric name.  I really want to see the style of the icon right away ☺️

**Comment by leandroberetta** [comment]
@pantianying I think a better solution would be to add the metric into your Prometheus, because that memory metric is the right one to use.

**Comment by jmazzitelli** [comment]
I think what @pantianying is saying is that he can change it in HIS code (he has a fork built for his production system) - if it will work.

**Comment by leandroberetta** [comment]
That is doable, you can change the metric, but you will also need to find another metric that you have available suitable for the istiod memory container. I mean, I think it's better to add the proper metric rather than add another one that needs to represent the same.

**Comment by leandroberetta** [comment]
I investigated a little bit more which other metrics we have, and I remember that Istio provides some dashboards for the control plane. There, I saw that they indicated both CPU and memory from a container perspective and a process perspective (a process, because what is really running is the process, the container is a mix between security and isolation).

These are the charts:

![image](https://github.com/kiali/kiali/assets/1286393/3094f4bf-28be-4de5-8df4-0ec3d45f295d)

So, Kiali today shows a combination (process cpu, and container memory). The following PR I created is for normalize this situation and show container cpu and container memory, but, we might show process related metrics in case we don't have the container ones (this is the case of OSSM when it's using it's own Prometheus).

We might want to show both values if we have them. 

Really, thinking in OSSM, I don't know how many customers will configure it to use OpenShift's Prometheus (where container metrics are available), so I guess we might want to add the process one too.
